exchanges/gateio/gateio_test.go

```diff
@@ -74,838 +71,2829 @@ func TestStart(t *testing.T) {
+	}
+}
+
+func TestSubmitOrder(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip()
+	}
+	var orderSubmission = &order.Submit{
+		Exchange: g.Name,
+		Pair: currency.Pair{
+			Delimiter: currency.UnderscoreDelimiter,
+			Base:      currency.BTC,
+			Quote:     currency.USDT,
+		},
+		Side:      order.Buy,
+		Type:      order.Limit,
+		Price:     1,
+		Amount:    1,
+		ClientID:  "meowOrder",
+		AssetType: asset.CrossMargin,
+	}
+	_, err := g.SubmitOrder(context.Background(), orderSubmission)
+	if err != nil {
+		t.Errorf("Order failed to be placed: %v", err)
+	}
+}
+
+func TestCancelExchangeOrder(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip()
+	}
+	var orderCancellation = &order.Cancel{
+		OrderID:       "1",
+		WalletAddress: core.BitcoinDonationAddress,
+		AccountID:     "1",
+		Pair:          currency.NewPair(currency.LTC, currency.BTC),
+		AssetType:     asset.Spot,
+	}
+	err := g.CancelOrder(context.Background(), orderCancellation)
+	if err != nil && !strings.Contains(err.Error(), "ORDER_NOT_FOUND") {
+		t.Errorf("%s CancelOrder error: %v", g.Name, err)
+	}
+}
+
+func TestCancelBatchOrders(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip()
+	}
+	currencyPair := currency.NewPair(currency.LTC, currency.BTC)
+	_, err := g.CancelBatchOrders(context.Background(), []order.Cancel{
+		{
+			OrderID:       "1",
+			WalletAddress: core.BitcoinDonationAddress,
+			AccountID:     "1",
+			Pair:          currencyPair,
+			AssetType:     asset.Spot,
+		}, {
+			OrderID:       "2",
+			WalletAddress: core.BitcoinDonationAddress,
+			AccountID:     "1",
+			Pair:          currencyPair,
+			AssetType:     asset.Spot,
+		}})
+	if err != nil && !strings.Contains(err.Error(), "ORDER_NOT_FOUND") {
+		t.Errorf("%s CancelOrder error: %v", g.Name, err)
+	}
+}
+
+func TestGetDepositAddress(t *testing.T) {
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	_, err := g.GetDepositAddress(context.Background(), currency.USDT, "", "TRX")
+	if err != nil {
+		t.Error("Test Fail - GetDepositAddress error", err)
+	}
+}
+
+func TestGetActiveOrders(t *testing.T) {
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	var getOrdersRequest = order.GetOrdersRequest{
+		Pairs:     []currency.Pair{currency.NewPair(currency.USDT, currency.BTC)},
+		Type:      order.AnyType,
+		AssetType: asset.Spot,
+	}
+	_, err := g.GetActiveOrders(context.Background(), &getOrdersRequest)
+	if err != nil {
+		t.Errorf(" %s GetActiveOrders() error: %v", g.Name, err)
+	}
+}
+
+func TestGetOrderHistory(t *testing.T) {
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	var getOrdersRequest = order.GetOrdersRequest{
+		Type:      order.AnyType,
+		AssetType: asset.Spot,
+	}
+	currPair := currency.NewPair(currency.LTC, currency.BTC)
+	currPair.Delimiter = currency.UnderscoreDelimiter
+	getOrdersRequest.Pairs = []currency.Pair{currPair}
+
+	_, err := g.GetOrderHistory(context.Background(), &getOrdersRequest)
```

Comment:
This doesn't work along with incorrect order side error the request path is wrong:

```
GateIO attempt 1 request path: https://api.gateio.ws/api/v4//api/v4//spot/my_trades?account=spot&currency_pair=LTC_BTC
```